### DISCUSSION TITLE
Errors found while using components in Next JS 14 project

### DISCUSSION THREAD

**Discussion by tomjackking** [WestpacGEL/GEL-next#253] [General]
Package.json (in case of any potential config issues):
<img width="135" alt="image" src="https://github.com/WestpacGEL/GEL-next/assets/46306101/cef7cec6-b9f8-42c3-93cd-9359bac29a8c">

**Autocomplete:**
When using Autocomplete.Item:
![image](https://github.com/WestpacGEL/GEL-next/assets/46306101/7e80e034-5d54-4e9d-b792-2a9e3da83c80)

**Alert:**
When attempting to override Icon through icon prop:
<img width="480" alt="image" src="https://github.com/WestpacGEL/GEL-next/assets/46306101/83c96120-3629-47d4-bec5-8763f650a655">

**Comment by samithaf** [comment]
Related to https://github.com/vercel/next.js/issues/51593

**Comment by tomjackking** [comment]
As for the issue in Alert, this will pop up whenever we use onClick (or other event handler) or pass an Icon (or other component) to the component. Even though the component export from GEL may have 'use client' at the top the page for the component you still can't do the previously mentioned if that component is in a server component.

A work around I have found (possibly even the solution) is to make the component you need, for example an Alert with a different Icon and onClose function inside the app with 'use client' at the top and import that into the server component:
<img width="358" alt="image" src="https://github.com/WestpacGEL/GEL-next/assets/46306101/7f988c31-ca50-4409-9200-44b2df25ea85">
<img width="324" alt="image" src="https://github.com/WestpacGEL/GEL-next/assets/46306101/ab53809c-64e2-40e3-9d40-ad84c82fb66a">

**Reply by tomjackking** [reply]
Alternate way to pass function from server component to client component:
<img width="326" alt="image" src="https://github.com/WestpacGEL/GEL-next/assets/46306101/873b34a6-deb7-4485-8163-95c465611400">
